pkg/river/internal/rivertags/rivertags.go

```diff
@@ -45,9 +45,9 @@ func (f Flags) GoString() string { return f.String() }
 
 // Field is a tagged field within a struct.
 type Field struct {
-	Name  string // Name of tagged field
-	Index []int  // Index into field (reflect.Value.FieldByIndex)
-	Flags Flags  // Flags assigned to field
+	Name  []string // Name of tagged field
```

Comment:
This needed to be changed into a slice since block names may be separated into fragments (e.g., `a.b.c` should be `[]string{"a", "b", "c"}` so that you only need to do the split once). This also helps with building the tree.